forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/RSASASignatureSigner.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.devicebind
+
+import com.nimbusds.jose.JOSEException
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.crypto.RSASSASigner
+import com.nimbusds.jose.crypto.impl.RSASSA
+import com.nimbusds.jose.util.Base64URL
+import java.security.InvalidKeyException
+import java.security.PrivateKey
+import java.security.Signature
+import java.security.SignatureException
+
+/**
+ * Extend of [RSASSASigner] to cache and expose the signature
+ */
+internal class RSASASignatureSigner : RSASSASigner {
+    var signature: Signature
+        private set
+
+    constructor(jwsAlgorithm: JWSAlgorithm , privateKey: PrivateKey) : super(privateKey) {
+        signature = getInitiatedSignature(jwsAlgorithm)
```

Comment:
if you provide the private key the signature value is same for both binding and signing scenarios?

Reply:
Implementation updated, only take Signature.